data-prepper-logstash-configuration/src/test/java/org/opensearch/dataprepper/logstash/mapping/GrokNamedCapturesUtilTest.java

```diff
@@ -0,0 +1,125 @@
+
+        for (final Map.Entry<String, String> patternDefinition : result.getMappedPatternDefinitions().entrySet()) {
+           assertThat(patternDefinition.getValue().equals(namedCapturesPattern), equalTo(true));
+           final String expectedResult = String.format("%%{%s:%s}", patternDefinition.getKey(), namedCapturesName);
+           assertThat(result.getMappedRegex().equals(expectedResult), equalTo(true));
+        }
+    }
+
+    @Test
+    public void testConnectedNamedCaptures() {
+        final String regex = String.format("%s(?<%s>%s)(?<%s>%s)",
+                randomPrefix, firstNamedCapturesName, firstNamedCapturesPattern,
+                secondNamedCapturesName, secondNamedCapturesPattern);
+
+        final GrokNamedCapturesPair result = GrokNamedCapturesUtil.convertRegexNamedCapturesToGrokPatternDefinitions(regex);
+        assertThat(result.getMappedPatternDefinitions().size(), equalTo(2));
+
+        int index = 0;
+        final List<String> patternDefinitionNames = new ArrayList<>();
+        for (final Map.Entry<String, String> patternDefinition : result.getMappedPatternDefinitions().entrySet()) {
+            assertThat(patternDefinition.getValue().equals(namedCapturesPatterns.get(index)), equalTo(true));
+            patternDefinitionNames.add(patternDefinition.getKey());
+            index++;
+        }
+        final String expectedResult = String.format("%s%%{%s:%s}%%{%s:%s}", randomPrefix, patternDefinitionNames.get(0), namedCapturesNames.get(0), patternDefinitionNames.get(1), namedCapturesNames.get(1));
+        assertThat(result.getMappedRegex().equals(expectedResult), equalTo(true));
+    }
+
+    @Test
+    public void testSeparatedNamedCaptures() {
+        final String regex = String.format("%s(?<%s>%s) %s (?<%s>%s)%s",
+                randomPrefix, firstNamedCapturesName, firstNamedCapturesPattern, randomMiddle,
+                secondNamedCapturesName, secondNamedCapturesPattern, randomSuffix);
+
+        final GrokNamedCapturesPair result = GrokNamedCapturesUtil.convertRegexNamedCapturesToGrokPatternDefinitions(regex);
+        assertThat(result.getMappedPatternDefinitions().size(), equalTo(2));
+
+        int index = 0;
+        final List<String> patternDefinitionNames = new ArrayList<>();
+        for (final Map.Entry<String, String> patternDefinition : result.getMappedPatternDefinitions().entrySet()) {
+            assertThat(patternDefinition.getValue().equals(namedCapturesPatterns.get(index)), equalTo(true));
+            patternDefinitionNames.add(patternDefinition.getKey());
+            index++;
+        }
+        final String expectedResult = String.format("%s%%{%s:%s} %s %%{%s:%s}%s", randomPrefix, patternDefinitionNames.get(0), namedCapturesNames.get(0), randomMiddle,
+                patternDefinitionNames.get(1), namedCapturesNames.get(1), randomSuffix);
+        assertThat(result.getMappedRegex().equals(expectedResult), equalTo(true));
```

Comment:
I find it better to use the following form because you get better output when the assertion fails:

```
assertThat(result.getMappedRegex(), equalTo(expectedResult));
```